lib/govuk_publishing_components/config.rb

```diff
@@ -29,7 +29,7 @@ def self.gem_directory
       Gem::Specification.find_by_name("govuk_publishing_components").gem_dir
     end
 
-    ALL_ASSETS = {
+    ALL_STYLESHEETS = {
```

Comment:
Thanks for the review and feedback @1pretz1! 👍

@MartinJJones and I have discussed this one previously and thought that listing all the stylesheets would a better way to replicate the manifest and Dart Sass initializer and by individually listing each stylesheet, it might provide an easier overview of the included stylesheets at a glance.

Also, excluding `_copy-to-clipboard.scss` is intentional since, as you mentioned, it’s an empty an stylesheet; thereby eliminating an unnecessary network request.

I agree it's not ideal and looping through the directory contents would be nicer (potentially less prone to errors as well). However, considering the reasons mentioned above, what are your thoughts on this?

Reply:
Ah good to know this has already been discussed and I see where you both are coming from. My view is that we probably want to avoid hardcoding a long list that we could just generate from a directory, as it means one less file to update when components are added / removed / renamed. The current list is also a bit unwieldy due to the amount of lines and their length.

I think as the list comprises of solely stylesheets from the components dir (+ application.scss) then it shouldn't be any harder to glance at that directory vs scan this list we have defined in the config file.

I've had a more refined attempt with the purpose of being as explicit as possible to convey the intent of the code - what do you think? I'm also happy also happy to be persuaded against a loop if you and Martin think an explicit list has more advantages than disadvantages.

```
gem_directory = GovukPublishingComponents::Config.gem_directory
component_stylesheets_to_include = "/app/assets/stylesheets/govuk_publishing_components/components/*.scss"
component_stylesheets_with_gem_dir = Dir.glob(gem_directory + component_stylesheets_to_include)
application_stylesheet_with_gem_dir = "#{gem_directory}/app/assets/stylesheets/component_guide/application.scss"

stylesheets_hash = { application_stylesheet_with_gem_dir => "component_guide/application.css" }

stylesheet_files_with_gem_dir.each_with_object(stylesheets_hash) do |file_path, hsh|
  hsh[file_path] = "govuk_publishing_components/components/" + File.basename(file_path)
end
```

Reply:
Thanks @1pretz1. Thinking about it some more you're right that managing it this way would be more manageable and we had concerns about including an extensive list of files, wondering if there might be a better way or a better place to manage them; we do have tests in place to ensure the correct number of stylesheets are generated.

I’ve included the code above and made a couple of small changes to:
- use the `gem_directory` method already included in the config file
- exclude `copy-to-clipboard` and
- changed the filename extension for each generated file to `css` (from `scss`).

Reply:
Awesome, all looks good to me and thanks for clarifying. Seems a bit confusing to maintain an empty stylesheet (+ the logic it creates in the method) but this isn't a concern of the PR 😄